### PR TITLE
fix drag and drop

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

File: src/glfw3.jl [hunk 1]
```diff
@@ -437,7 +437,7 @@ SetCursor(window::Window, cursor::Cursor) = ccall( (:glfwSetCursor, lib), Void,
 @callback CursorPos(window::Window, xpos::Cdouble, ypos::Cdouble)
 @callback CursorEnter(window::Window, entered::Cint) -> (window, @compat Bool(entered))
 @callback Scroll(window::Window, xoffset::Cdouble, yoffset::Cdouble)
-@callback Drop(window::Window, count::Cint, paths::Ptr{Ptr{Cchar}}) -> (window, map(bytestring, pointer_to_array(paths, count)))
+@callback Drop(window::Window, count::Cint, paths::Ptr{Ptr{Cchar}}) -> (window, map(x->utf8(Ptr{UInt8}(x)), pointer_to_array(paths, count)))
```
Comment (jayschwa) [inline src/glfw3.jl]: The `Ptr{Cchar}` to `Ptr{UInt8}` cast might be avoidable if `paths` is changed to `Ptr{Ptr{IUint8}}`. Did you try that already?

Comment (SimonDanisch) [inline src/glfw3.jl]: No, but that could work ;) I was a little bit in a hurry...

**Comment by jayschwa** [comment]
@SimonDanisch were you experiencing a problem with `bytestring` that `utf8` fixed, or did you want to change it for correctness? I played around with this on my dev branch and found that `utf8` is unable to accept either `Ptr{Cchar}` or `Cstring`, which requires changing the `ccall` types from how GLFW documents them. The [documentation for calling C](http://docs.julialang.org/en/release-0.4/manual/calling-c-and-fortran-code/) still recommends `bytestring` and it seemed to handle UTF-8 encoded strings without problems when I tested.

**Comment by SimonDanisch** [comment]
Bytestring is not type stable so it returns different string types depending on the encoding, which I find slightly annoying. Since the documentation said it's UTF8 encoded, I thought it makes sense to always return an UTF8String.

**Comment by jayschwa** [comment]
Filed https://github.com/JuliaLang/julia/issues/13669

I agree that `utf8` would be better for the sake of explicitness, but not if it requires casting or less explicit types elsewhere (i.e. not using `Cstring`)

**Comment by SimonDanisch** [comment]
Great, you're going for the thorough fix :)

**Comment by jayschwa** [comment]
Obsoleted by #90.

**Pull request closed without merging**